compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1466,6 +1467,10 @@ private void checkForExportableType(BTypeSymbol symbol, DiagnosticPos pos) {
                 }
                 checkForExportableType(invokableType.retType.tsymbol, pos);
                 return;
+            case TypeTags.PARAMETERIZED_TYPE:
```

Comment:
Was there something specific that failed, requiring this change?

Reply:
Yes actually. I was getting an error for trying to export a non public symbol for something like 
```function test(typedesc<object {}> T) returns T {}```

Reply:
Had an offline discussion with @dulvinw regarding this too, previously `typedesc<object {}> T` in this same example was marked as a public param, which is why we didn't get the error.

We will go ahead with this change.